File: leetcode/easy/1._two_sum/mac.js
// https://leetcode.com/problems/two-sum
// Runtime: 344 ms, faster than 5.06% of JavaScript online submissions for Two Sum.
// Memory Usage: 35.6 MB, less than 18.60% of JavaScript online submissions for Two Sum.

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numsLen = nums.length;
    
    for (let i = 0; i < numsLen; i++) {
        const val = target - nums[nums.indexOf(parseInt(target) - parseInt(nums[i]))];
        
        for (let j = 0; j < numsLen; j++) {
            if (i !== j) {
                if (val + nums[j] === target) {
                    return [i, j];
                }
            }
        }
    }
    
};

File: leetcode/easy/1._two_sum/mac.swift
https://leetcode.com/problems/two-sum/
Runtime: 32 ms, faster than 95.20% of Swift online submissions for Two Sum.
Memory Usage: 20.9 MB, less than 5.88% of Swift online submissions for Two Sum.

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dic = [Int: Int]()

        for index in 0..<nums.count {
            let val = nums[index]
            if let before = dic[target - val] {
                return [before, index]
            }

            dic[val] = index
        }

        return [0, 0]
    }
}